opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/async/instrument/TracedAsyncConfigurer.java

```diff
@@ -0,0 +1,33 @@
+package io.opentracing.contrib.spring.cloud.async.instrument;
+
+import java.util.concurrent.Executor;
+
+import org.springframework.aop.interceptor.AsyncUncaughtExceptionHandler;
+import org.springframework.scheduling.annotation.AsyncConfigurer;
+import org.springframework.scheduling.annotation.AsyncConfigurerSupport;
+
+import io.opentracing.Tracer;
+
+/**
+ * @author kameshsampath
+ */
+public class TracedAsyncConfigurer extends AsyncConfigurerSupport {
+
+    private final Tracer tracer;
+    private final AsyncConfigurer delegate;
+
+    public TracedAsyncConfigurer(Tracer tracer, AsyncConfigurer delegate) {
+        this.tracer = tracer;
+        this.delegate = delegate;
+    }
+
+    @Override
+    public Executor getAsyncExecutor() {
+       return new TracedExecutor(this.tracer, this.delegate.getAsyncExecutor());
+    }
+
+    @Override
+    public AsyncUncaughtExceptionHandler getAsyncUncaughtExceptionHandler() {
```

Comment:
Might be worth defining tests for uncaught exceptions to make sure spans are finished correctly - and possibly have the exception logged in the span?

Reply:
+1,

Reply:
This is related to Executor instrumentation, it does not create any spans, it just propagate parent span to the submitted thread. We cannot finish this span it's the responsibility of the span creator. 

I have added exception test and logs to the `Async` annotation use-case.